lib/plugins/inventory.js

```diff
@@ -68,29 +68,39 @@ function inject (bot, { version, hideErrors }) {
     }
   })
 
-  function consume (cb) {
+  async function consume () {
     if (lastEating) {
       lastEating(new Error('Consuming cancelled due to calling bot.consume() again'))
     }
 
     if (bot.food === 20) {
-      return cb(new Error('Food is full'))
+      throw new Error('Food is full')
     }
 
     activateItem()
 
-    lastEating = (err) => setImmediate(cb, err)
+    return new Promise((resolve, reject) => {
+      lastEating = (err) => {
+        if (err) reject(err)
+        else resolve()
+      }
+    })
```

Comment:
Can be rewritten as following to ommit new promise:
```js
let eatingState = false

bot._client.on('entity_status', (packet) => {
  if (packet.entityId === bot.entity.id && packet.entityStatus === 9 && eatingState) {
    bot.emit('eatingDone')
  }
})

async function consume () {
  if (eatingState) {
    throw ...
  }

  if (bot.food === 20) {
    throw ...
  }

  eatingState = true
  await once(bot, 'eatingDone')
  eatingState = false
}
```

Reply:
That's actually what we've been discussing quite a while ago on discord when I was working on #984. Same applies for a couple of other methods

Reply:
Yeah, I've been going around and fixing those, I'll keep doing that

Reply:
Hmm though looking at this specific example, I'm not sure if it's correct. If consume() is called while another is running, the _other_ consume should be rejected, not the new one.

Reply:
You have the overall idea though

Reply:
`once` fails when `error` emitted so you'd have to do a new emitter for that. Both consume and fish can share the emitter:
```js
const state = new EventEmitter()
state.eating = false

bot._client.on('entity_status', (packet) => {
  if (packet.entityId === bot.entity.id && packet.entityStatus === 9 && state.eating) {
    state.emit('eatingDone', true)
  }
})

async function consume () {
  if (state.eating) {
    state.emit('eatingDone', false)
  }

  if (bot.food === 20) {
    throw ...
  }

  state.eating = true
  if (!await once(state, 'eatingDone')) {
    throw ...
  }
  state.eating = false
}
```